llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -8388,6 +8388,24 @@ static SelectPatternResult matchMinMax(CmpInst::Predicate Pred,
     }
   }
 
+  // (X > Y) ? X : (Y - 1) ==> MIN(X, Y - 1)
+  // (X < Y) ? X : (Y + 1) ==> MAX(X, Y + 1)
+  // When overflow corresponding to the sign of the comparison is poison.
+  // Note that the UMIN case is not possible as we canonicalize to addition.
+  if (CmpLHS == TrueVal) {
+    if (Pred == CmpInst::ICMP_SGT &&
+        match(FalseVal, m_NSWAddLike(m_Specific(CmpRHS), m_ConstantInt<1>())))
```

Comment:
```suggestion
        match(FalseVal, m_NSWAddLike(m_Specific(CmpRHS), m_One())))
```
Please use `m_One` and `m_AllOnes`. Can you add some vector tests?

Reply:
Fixed! and I've added vector tests for each case.